src/shapes/Text/StyledText.ts

```diff
@@ -243,11 +253,15 @@ export abstract class StyledText<
   }
 
   /**
-   * get the reference, not a clone, of the style object for a given character,
-   * if not style is set for a pre det
+   * Get a reference, not a clone, to the style object for a given character,
+   * if no style is set for a line or char, return a new empty object.
+   * This is tricky and confusing because when you get an empty object you can't
+   * determine if it is a reference or a new one.
+   * @TODO this should always return a reference or always a clone or undefined when necessary.
```

Comment:
good comment
bad design

Reply:
yes we have a TODO to fix it. Fixing it now has a lot of subtle consequences i don't want to think of